proposals/model-management.md

```diff
@@ -0,0 +1,149 @@
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-405B-Instruct"
+  cacheProfile: HyperDisk
+
+# A Model without any cache specified will use the default set in KubeAI's config.yaml
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+
+# Model that explicitly wont use a cache
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+  cacheProfile: None
+```
+
+### Finetuning Example
+
+If a user submits a finetuning request (`POST /v1/fine_tuning/jobs`), they have the option to specify a location to store the adapters:
+
+```json
+# OpenAI-compatible finetune job request
+{
+  # Model.metadata.name
+  "model": "llama-3.1-8b",
+  # KubeAI-Only: Training file could be specified as URL: "s3://..."
+  "training_file": "<id-of-file-uploaded-via-files-api>",
+  # KubeAI-Only: Location to store parameters
+  # Alternative: "hf://some-user/some-repo"
+  # If not specified, default location determined from KubeAI config.yaml.
+  "storage_url": "s3://some-bucket/some-dir/"
+}
+```
+
+After this job completes the base Model would have an adapter appended:
+
+```yaml
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+  adapters:
+  # +Adapter:
+  - name: "name-of-finetune-job"
+    url: "s3://some-bucket/some-dir/name-of-finetune-job"
+```
+
+Users can specify an adapter by name at inference time.
+
+### KubeAI Config
+
+Model cache profiles and finetune default storage locations should be defined at the KubeAI `config.yaml` level:
+
+```yaml
+finetuning:
+  storage:
+    defaultURLTemplate: "s3://my-bucket/finetuning/{model}/{job}"
+  
+models:
+  cache:
+    defaultProfile: Bucket # Could also be "None"
+    profiles:
+      # Admins can define their own profile settings (and profile names):
+      StandardBucket:
+        bucket:
+          url: s3://my-bucket/models/
+      EFS:
+        sharedFilesystem:
+          storageClass: aws-efs
+          mode: SharedFilesystem
+          baseDir: /models/
+      HyperDisk:
+        staticVolume:
+          storageClass: gcp-hyperdisk
+```
+
+### Cache Options
+
+#### Bucket Storage
+
+This mode relies on access to a bucket.
+
+To preserve local deployment as an option this bucket could be provided by a Helm-deployed instance of Minio (possible License issue?), OR [SeaweedFS](https://github.com/seaweedfs/seaweedfs) (mature/secure?).
+
+![Bucket Mode](./diagrams/model-mgmt-buckets.excalidraw.png)
+
+#### Volume Storage
+
+##### Shared Filesystem Storage
+
+This mode relies on a Kubernetes PersistentVolumes for storage.
+
+- How it works: always mounted as ReadWriteMany
+- Examples:  - NFS, EFS, GCP Filestore
+- NOTEs:
+  - Works for models and adapaters.
+  - Typically provisioned outside of the k8s API
+
+##### Static Volume Storage
+
+This mode relies on a Kubernetes PersistentVolumes for storage.
+
+- How it works: mounted as ReadWriteOnce to initialize, then ReadOnlyMany for serving
+- Examples: GCP Hyperdisk ML
```

Comment:
GCP Persistent Disk ReadManyOnly is another option

Reply:
Good callout - will add that in but mention the max-attachment limitation